src/main/java/jarvis/logic/commands/AddNotesCommand.java

```diff
@@ -81,11 +83,21 @@ public CommandResult execute(Model model) throws CommandException {
 
     @Override
     public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof AddNotesCommand // instanceof handles nulls
-                && lessonIndex.equals(((AddNotesCommand) other).lessonIndex)
-                && studentIndex != null && ((AddNotesCommand) other).studentIndex != null
-                && studentIndex.equals(((AddNotesCommand) other).studentIndex)
-                && notes.equals(((AddNotesCommand) other).notes));
+        if (other == this) { // short circuit if same object
+            return true;
+        }
+
+        if (!(other instanceof AddNotesCommand)) { // instanceof handles nulls
+            return false;
+        }
+
+        AddNotesCommand otherAddNotes = (AddNotesCommand) other;
+
+        if (studentIndex == null || otherAddNotes.studentIndex == null) {
+            return studentIndex == null && otherAddNotes.studentIndex == null;
```

Comment:
This skips the equality check for `lessonIndex` and `notes` since it returns immediately.